addons/addon-base-raas/packages/base-raas-services/lib/data-egress/__test__/data-egress-service.test.js

```diff
@@ -0,0 +1,216 @@
+        expect.objectContaining({ boom: true, code: 'forbidden', safe: true }),
+      );
+    });
+
+    it('should create egress store with egress store feature enabled', async () => {
+      const s3Policy = testS3PolicyFn();
+      dataEgressService._settings = {
+        get: settingName => {
+          if (settingName === 'enableEgressStore') {
+            return true;
+          }
+          if (settingName === 'egressStoreKmsKeyAliasArn') {
+            return 'test-egressStoreKmsKeyAliasArn';
+          }
+          if (settingName === 'egressStoreBucketName') {
+            return 'test-egressStoreBucketName';
+          }
+          return undefined;
+        },
+      };
+      const requestContext = {};
+      const rawEnvironment = {
+        id: 'test-id',
+        name: 'test-raw-environment-name',
+        createdBy: 'test-raw-environment-createdby',
+        updatedBy: 'test-updatedBy',
+        projectId: 'test-raw-environment-projectId',
+        rev: 0,
+        inWorkflow: 'test-inWorkflow',
+        status: 'test-status',
+        cidr: '0.0.0.0',
+        createdAt: 'testCreatedAt',
+        updatedAt: 'testUpdatedAt',
+        studyIds: [],
+        indexId: 'testIndexId',
+        description: 'testDescription',
+        envTypeConfigId: 'envTypeConfigId',
+        envTypeId: 'envTypeId',
+        hasConnections: true,
+      };
+
+      const mockEgressStoreInfo = {
+        id: `egress-store-${rawEnvironment.id}`,
+        readable: true,
+        writeable: true,
+        kmsArn: 'test-arn',
+        bucket: 'test-egressStoreBucketName',
+        prefix: 'test-id/',
+        envPermission: {
+          read: true,
+          write: true,
+        },
+        status: 'reachable',
+        createdBy: rawEnvironment.createdBy,
+        workspaceId: rawEnvironment.id,
+        projectId: rawEnvironment.projectId,
+        resources: [
+          {
+            arn: `arn:aws:s3:::test-egressStoreBucketName/${rawEnvironment.id}/`,
+          },
+        ],
+      };
+      AWSMock.mock('KMS', 'describeKey', (params, callback) => {
+        expect(params).toMatchObject({
+          KeyId: 'test-egressStoreKmsKeyAliasArn',
+        });
+        callback(null, { KeyMetadata: { Arn: 'test-arn' } });
+      });
+
+      AWSMock.mock('S3', 'getBucketPolicy', (params, callback) => {
+        expect(params).toMatchObject({
+          Bucket: 'test-egressStoreBucketName',
+        });
+        callback(null, { Policy: JSON.stringify(s3Policy) });
+      });
+
+      const putBucketPolicyMock = jest.fn((params, callback) => {
+        expect(params).toMatchObject({
+          Bucket: 'test-egressStoreBucketName',
+        });
+        callback(null, {});
+      });
+      AWSMock.mock('S3', 'putBucketPolicy', putBucketPolicyMock);
+
+      const result = await dataEgressService.createEgressStore(requestContext, rawEnvironment);
+      expect(putBucketPolicyMock).toHaveBeenCalledTimes(1);
```

Comment:
Where are we verifying that the correct policy was being updated?

Reply:
Tests added in commit: c0dfd0b9e97060702c0940a873a3f18d29941584